File: September/12th September,2024/Leetcode(1684).java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        HashMap<Character,Integer>mp=new HashMap<>();
        for(int i=0;i<allowed.length();i++){
            mp.put(allowed.charAt(i),1);
        }
        int ans=0;
        for(int i=0;i<words.length;i++){
            boolean flag=true;
            for(int j=0;j<words[i].length();j++){
                char ch=words[i].charAt(j);
                if(!mp.containsKey(ch)){
                    flag=false;
                    break;
                }
                    
            }
            if(flag) ans+=1;
        }
        return ans;
    }
}

File: September/12th September,2024/Leetcode(1684).js
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
var countConsistentStrings = function (allowed, words) {
    let s = new Set(allowed)
    let ans = 0
    for (let i = 0; i < words.length; i++) {
        let flag = true;
        const w = words[i];
        for (let j = 0; j < w.length; j++) {
            const ch = w[j]
            if (!s.has(ch)) {
                flag = false;
                break
            }
        }
        if (flag) ans++;
    }
    return ans;
};